### PR TITLE
Script updates for astc-encoder-3.2 on Linux aarch64 (ARM64)

### PR DESCRIPTION
**Pull request by spham-amzn**
**Build Command**
```
python3 ../3p-package-scripts/o3de_package_scripts/build_package.py --search_path . astc-encoder-3.2-rev2-linux-aarch64
```

**Build output**
```
Creating/Updating package: astc-encoder-3.2-rev2-linux-aarch64
    Adding files to: "./packages/temp/temp_package_astc-encoder-3.2-rev2-linux-aarch64"
    Created: ./packages/astc-encoder-3.2-rev2-linux-aarch64.tar.xz
    Created: ./packages/astc-encoder-3.2-rev2-linux-aarch64.tar.xz.SHA256SUMS
    Created: ./packages/astc-encoder-3.2-rev2-linux-aarch64.tar.xz.content.SHA256SUMS
    Created: ./packages/astc-encoder-3.2-rev2-linux-aarch64.PackageInfo.json
    - Validating package: astc-encoder-3.2-rev2-linux-aarch64 in folder ./packages....
    Package Hash: ce8bac6283db84341e5cd51907d85e3da6c0b6fe980dff8112505ea7788752ed *astc-encoder-3.2-rev2-linux-aarch64.tar.xz
```

**Upload output**
```
Using bucket: ly-dev-3p-packages
Package: astc-encoder-3.2-rev2-linux-aarch64 ...
    - Validating package: astc-encoder-3.2-rev2-linux-aarch64 in folder ./packages....
    - Uploading astc-encoder-3.2-rev2-linux-aarch64.tar.xz...
    - Uploading astc-encoder-3.2-rev2-linux-aarch64.tar.xz.SHA256SUMS...
    - Uploading astc-encoder-3.2-rev2-linux-aarch64.tar.xz.content.SHA256SUMS...
    - Uploading astc-encoder-3.2-rev2-linux-aarch64.PackageInfo.json...
    - Uploaded package astc-encoder-3.2-rev2-linux-aarch64.

```

### CODE REVIEW

**Review by lawsonamzn — APPROVED** [review]
looks okay, but in genereal (this applies to all the packages) I worry about managing updates for this in the future though if you need a arm64 linux to update them.    I take it theres no great way to cross compile or specify a cross target for all of these?

**Comment by spham-amzn** [comment]
> looks okay, but in genereal (this applies to all the packages) I worry about managing updates for this in the future though if you need a arm64 linux to update them. I take it theres no great way to cross compile or specify a cross target for all of these?

Its possible to cross-compile, but it becomes much more involved especially given that there are a lot different build systems that need to be configured in their own way to support this.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: package-system/astc-encoder/build_linux.sh [hunk 1]
```diff
@@ -5,10 +5,22 @@
 #
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 
-# Note: on x86/x64 platforms, O3DE requires a minimum of SSE 4.1, so we do request this.
-cmake -S temp/src -B temp/build -G "Unix Makefiles" \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DISA_SSE41=ON \
-    || exit $?
+CPU_ARCHITECTURE=$(lscpu | grep Architecture | awk '{print $2}')
```
[the remark this answers is not in the document]
Comment (spham-amzn) [reply package-system/astc-encoder/build_linux.sh:8]: turns out i can't rely on /proc/cpuinfo. The result does not appear to be standard. On the ubuntu aarch64 EC run, the 'architecture' is a number
```
processor       : 0
BogoMIPS        : 243.75
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x3
CPU part        : 0xd0c
CPU revision    : 1
```
While on ubuntu x86_64 on a thread ripper:
```
gistfile1.txt
processor	: 61
vendor_id	: AuthenticAMD
cpu family	: 23
model		: 49
model name	: AMD Ryzen Threadripper 3970X 32-Core Processor
stepping	: 0
microcode	: 0x8301039
cpu MHz		: 2200.000
cache size	: 512 KB
physical id	: 0
siblings	: 64
core id		: 29
cpu cores	: 32
apicid		: 59
initial apicid	: 59
fpu		: yes
fpu_exception	: yes
cpuid level	: 16
wp		: yes
```

**Review by lawsonamzn — APPROVED** [review]
An alternative would be to pass it in from the package build config json file or even the host json file... perhaps knowing its a 64 bit arch at the pull_and_build_from_git level would help with cross compiling in the future anwyay

**Pull request merged**